Sources/PenguinStructures/Lens.swift

```diff
@@ -0,0 +1,42 @@
+//******************************************************************************
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/// `KeyPath`s with a statically known `Value` endpoint.
+// This protocol is only needed because we can't represent `: KeyPath<_,V>` as a constraint in the
+// type system.
+public protocol KeyPathProtocol: AnyKeyPath {
```

Comment:
Is there a better name here?

Reply:
No 😄 If you are uncomfortable with something in the code you're reviewing it's nice to provide an alternative suggestion, yeah?

It is one of those things working around the fact that we can't use "is-a `KeyPath<_, _>`" as a constraint.